### PR TITLE
Add extra options to helm chart

### PR DESCRIPTION
**Pull request by blackjid**
This options has been useful for the way we have being using this 
project

- add helm option to change the frontend service name
- add helm option to set varnish container lifecycle options
- add helm option to add extracontainers to the pod
- add helm option to set the pod termination grace period

### CODE REVIEW

**Comment by martin-helmich** [comment]
@blackjid I did my best to resolve the conflicts that occurred after I merged #71 -- but maybe you should have another look as well. 😉

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -123,6 +127,9 @@ spec:
             {{- toYaml .Values.exporter.resources | nindent 12 }}
           {{- end }}
         {{- end }}
+        {{- if .Values.extraContainers }}
+          {{- toYaml .Values.extraContainers | nindent 10 }}
```
Comment (martin-helmich) [inline chart/templates/deployment.yaml:131]: Is the indentation correct? If `.Values.extraContainers` is a list, this should be indented by 8 spaces if I'm not mistaken.

```suggestion
          {{- toYaml .Values.extraContainers | nindent 8 }}
```

EDIT, which would incidentally be the same indentation level as already used in `statefulset.yaml`.

File: chart/values.yaml [hunk 2]
```diff
@@ -93,7 +103,14 @@ initContainers: {}
...
 #     - name: template
 #       mountPath: /etc/varnish/tmpl
-#
+
+extraContainers: []
+# - name: prometheus-exporter
+#   image: jonnenauha/prometheus_varnish_exporter
+#   imagePullPolicy: IfNotPresent
+#   command:
+#   - prometheus_varnish_exporter
```
Comment (martin-helmich) [inline chart/values.yaml:112]: I think, even after #71 there's still a use case for adding extra containers (varnishncsa logging, for example), but #71 kind of makes this specific example obsolete.

**Review by martin-helmich — CHANGES_REQUESTED** [review]

File: chart/values.yaml [hunk 3]
```diff
@@ -24,6 +24,8 @@ configmap:
 
 # kube-httpcache specific configuration
 cache:
+  # name of backend service
+  # frontendService: kube-httpcache-headless
```
Comment (martin-helmich) [inline chart/values.yaml:28]: Just a quick thought: When supplying your own frontend service, is the service created by the chart itself still needed? OTOH, it probably wouldn't do any harm to just have it created anyway.

Open to suggestions. :shrug:

Comment (blackjid) [reply chart/values.yaml:28]: I'm still using the service created by the chart.

I'm following these approaches to get working readiness probes

https://github.com/mittwald/kube-httpcache/issues/36#issuecomment-713242670
https://github.com/mittwald/kube-httpcache/issues/36#issuecomment-740278034

File: chart/values.yaml (hunk 2 above)
Comment (blackjid) [reply chart/values.yaml:112]: 👍🏼 I'll leave a generic example

**Comment by blackjid** [comment]
I fixedup and rebase the fixes...

**Comment by blackjid** [comment]
hey @martin-helmich just bringing it up in case you have time :)

File: chart/values.yaml (hunk 3 above)
Comment (martin-helmich) [reply chart/values.yaml:28]: OK, I see. Nevermind, then. 😉

**Review by martin-helmich — APPROVED** [review]
Looking good. :+1:  Sorry for taking so long.

**Comment by blackjid** [comment]
No problem!, Thanks for the taking the time :)

**Pull request merged**